File: examples/nodejs/simple/example_3.js
const MdisClient = require("./src/index");

async function main() {
  try {
    let m = MdisClient.connect((host = "127.0.0.1"), (port = 6411));
    const resp = await m.set("token", "123456", 2);

    console.log("resp:", resp);

    setTimeout(async function () {
      const myToken = await m.get("token");

      console.log("token:", myToken);
    }, 1000 * 3);
    // console.log("completed");
  } catch (error) {
    console.log(error);
  }
}
main();

File: examples/rust/simple/examples/example_3.rs
//! Example 3: SET with a per-key expiration, then GET after it lapses.
use mdis_client_example::MdisClient;
use std::thread::sleep;
use std::time::Duration;

fn main() {
    let client = MdisClient::connect("127.0.0.1", 6411);

    match client.set("token", "123456", 2) {
        Ok(resp) => println!("resp: {resp}"),
        Err(e) => {
            println!("Error: {e}");
            return;
        }
    }

    sleep(Duration::from_secs(3));

    match client.get("token") {
        Ok(token) => println!("token: {token}"),
        Err(e) => println!("Error: {e}"),
    }
}
